test/test-dom.ts

```diff
@@ -40,6 +40,31 @@ test('should handle empty ast in quirks mode', (t) => {
   t.is(printDoc(doc), '');
 });
 
+test('should handle tree under ZERO_LENGTH node', (t) => {
+  const ast: TreeProto = {
+    quirks_mode: true,
+    root: 0,
+    tree: [
+      {
+        tagid: getTagId('ZERO_LENGTH'),
+        value: undefined,
+        attributes: [],
+        children: [
+          {
+            value: 'html',
+            tagid: getTagId('html'),
+            attributes: [],
+            children: [],
```

Comment:
```suggestion
```

Reply:
will double check if our current types allow for this

Reply:
https://github.com/ampproject/bento-compiler/blob/402a701b458353442656db75aadfbe3233abcb77/src/ast.ts#L32-L37

Reply:
I'll research more as to the origin of ZERO_LENGTH

Reply:
ack. I checked and the translator does produce empty arrays rather than `undefined`. Maybe not ideal esp on larger trees from a performance perspective with all the extra allocations, but negligible enough to ignore for now